src/pubsub/subscribe.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const ndjson = require('iterable-ndjson')
+const explain = require('explain-error')
+const bs58 = require('bs58')
+const { Buffer } = require('buffer')
+const { objectToQuery } = require('../lib/querystring')
+const configure = require('../lib/configure')
+const { ok, toIterable } = require('../lib/fetch')
+const SubscriptionTracker = require('./subscription-tracker')
+
+module.exports = configure(({ fetch, apiAddr, apiPath, headers }) => {
+  const subsTracker = SubscriptionTracker.singleton()
+
+  return async (topic, handler, options) => {
+    options = options || {}
+    options.signal = subsTracker.subscribe(topic, handler, options.signal)
+
+    const qs = objectToQuery({
+      arg: topic,
+      discover: options.discover,
+      ...(options.qs || {})
+    })
+
+    const url = `${apiAddr}${apiPath}/pubsub/sub${qs}`
+    let res
+
+    try {
+      res = await ok(fetch(url, {
+        method: 'POST',
+        signal: options.signal,
+        headers: options.headers || headers
+      }))
+    } catch (err) { // Initial subscribe fail, ensure we clean up
+      subsTracker.unsubscribe(topic, handler)
+      throw err
+    }
+
+    // eslint-disable-next-line no-console
+    const onError = options.onError || (err => console.error(err))
+
+    ;(async () => {
```

Comment:
Is this idiomatic JS now? (sorry, async/await beginner here)

Reply:
Probably not, I'll pull this out as a function that gets called to make it a bit more readable.